docker/iceberg-rest-fixture/Dockerfile

```
@@ -36,6 +39,10 @@ ENV CATALOG_JDBC_USER=user
 ENV CATALOG_JDBC_PASSWORD=password
 ENV REST_PORT=8181
 
+# Healthcheck for the iceberg rest service
+HEALTHCHECK --start-period=30s --retries=3 \
+  CMD curl --fail http://localhost:$REST_PORT || exit 1
```

I know that the JVM has a bit of a slow start, but 30 seconds seems a bit excessive for me. I would just start probing right away, and do this every second, for up to 10 seconds.
```suggestion
HEALTHCHECK --retries=10 --interval=1s \
  CMD curl --fail http://localhost:$REST_PORT/v1/config || exit 1
```

Reference: https://docs.docker.com/reference/dockerfile/#healthcheck

Checked this locally:
```
➜  iceberg-python git:(fd-use-load-catalog) curl --fail http://localhost:8181      
curl: (22) The requested URL returned error: 400
➜  iceberg-python git:(fd-use-load-catalog) curl --fail http://localhost:8181/v1/config
{"defaults":{},"overrides":{}}%                 
```